### PR TITLE
Allow passing of title to default error page

### PR DESCRIPTION
**Pull request by icco**

### CODE REVIEW

**Comment by lfades** [comment]
@icco Use case for this ? you can create your own `_error` page instead of trying to customize our default one, this would also have to be documented

**Comment by icco** [comment]
Sure @lfades, right now, if you pass a statusCode to next/error that isn't defined in the hash (for example, a 403), you get the error message "An unexpected error has occurred." As this isn't true, I figured updating your default error page to allow for correct messaging would be easier for lots of folks instead of recreating the page with a larger hash.

Screen Shot 2019-05-24 at 7 18 52 PM

File: packages/next/pages/_error.js [hunk 1]
```diff
@@ -14,12 +14,13 @@ export default class Error extends React.Component {
...
       res && res.statusCode ? res.statusCode : err ? err.statusCode : 404
-    return { statusCode }
+    
+    const title = statusCodes[statusCode] || 'An unexpected error has occurred'
+    return { statusCode, title }
   }
 
   render () {
-    const { statusCode } = this.props
-    const title = statusCodes[statusCode] || 'An unexpected error has occurred'
```
Comment (timneutkens) [inline packages/next/pages/_error.js]: Could you do `const title = this.props.title || statusCodes[statusCode] || 'An unexpected error has occurred'`

Comment (icco) [reply packages/next/pages/_error.js]: Sure

File: packages/next/pages/_error.js [hunk 2]
```diff
@@ -14,12 +14,14 @@ export default class Error extends React.Component {
   static getInitialProps ({ res, err }) {
     const statusCode =
       res && res.statusCode ? res.statusCode : err ? err.statusCode : 404
-    return { statusCode }
+    
+    const title = ''
```
Comment (timneutkens) [inline packages/next/pages/_error.js]:
```suggestion
```

**Pull request merged**